feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/validation/validations/SwapEnoughLiquidityValidation.kt

```diff
@@ -0,0 +1,22 @@
+package io.novafoundation.nova.feature_swap_impl.domain.validation.validations
+
+import io.novafoundation.nova.common.validation.ValidationStatus
+import io.novafoundation.nova.common.validation.validOrError
+import io.novafoundation.nova.feature_swap_api.domain.model.SwapQuote
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidation
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidationFailure
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidationFailure.NotEnoughLiquidity
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidationPayload
+
+class SwapEnoughLiquidityValidation(
+    private val quoteRetriever: suspend (SwapValidationPayload) -> Result<SwapQuote>
+) : SwapValidation {
+
+    override suspend fun validate(value: SwapValidationPayload): ValidationStatus<SwapValidationFailure> {
+        val quoteResult = quoteRetriever(value)
+
+        return validOrError(quoteResult.isSuccess) {
+            NotEnoughLiquidity
+        }
```

Comment:
Btw we should also add a separate screen state when we receive failed quote on the screen itself - currently it only hides all fields but it also needs to show a red alert you've added to layout and disable the button. You can do it in PR with field validations

Reply:
Okay, I made it in [this pr](https://github.com/novasamatech/nova-wallet-android/pull/1191)